File: boj/BinarySearch/BOJ_2003.py
# 백준 2003 수들의 합 2
# 21.02.13

import sys
from functools import cmp_to_key
from collections import deque
from itertools import combinations
from itertools import permutations
sys.setrecursionlimit(10**6)

n, m = map(int, sys.stdin.readline().split())
a = list(map(int, sys.stdin.readline().split()))
ans = 0

l = r = 0
temp = 0
while l < n:
    if r == n:
        temp -= a[l]
        l += 1
    elif temp < m:
        temp += a[r]
        r += 1
    elif temp >= m:
        temp -= a[l]
        l += 1
    
    if temp == m: ans += 1

print(ans)

File: boj/TwoPointer/BOJ_2003.swift
//  백준 2003 수들의 합 2
//  21.02.17

import Foundation

let n = readLine()!.split(separator: " ").map { Int($0)! }
let arr = readLine()!.split(separator: " ").map { Int($0)! }

var p1 = 0
var p2 = 0
var ans = 0
var temp = 0

while p1 < n[0] {
  if p2 == n[0] {
    temp -= arr[p1]
    p1 += 1
  } else if temp < n[1] {
    temp += arr[p2]
    p2 += 1
  } else {
    temp -= arr[p1]
    p1 += 1
  }
  if temp == n[1] { ans += 1 }
}

print(ans)
